beater/beater.go

```diff
@@ -163,6 +163,11 @@ func (bt *beater) run(ctx context.Context, cancelContext context.CancelFunc, b *
 		defer tracer.Close()
 	}
 
+	// add deprecation warning if running on a 32-bit system
+	if runtime.GOARCH == "386" {
+		bt.logger.Warn("deprecated target architecture: 32-bit systems support will be removed")
```

Comment:
`deprecation notice: 32-bit systems target architecture support will be removed`